### DISCUSSION TITLE
`too_many_connections` limit

### DISCUSSION THREAD

**Discussion by WasinWatt** [supabase/supabase#11410] [Questions] [answered]
When a client on my production site tries to subscribe realtime, it sometimes get `too_many_connections` which it said on the doc that too many connections are happening concurrently. However this is my production site and I want all users to be able to use my site. How can I get rid of this limit ? It's very low.

**Comment by GaryAustin1** [comment]
See the realtime pricing/limits section here:
https://supabase.com/pricing

You need to close connections that are not in use, if you can, to reduce the number active.  Otherwise need to contact support or adjust your plan.

**Reply by WasinWatt** [reply]
Yes I'm on the Pro plan and it says on the pricing page that it would charge $10 for extra 1000 peak concurrent connections which I'm fine with that. However, it doesn't seem to surpass the 500 connections limit.

**Reply by GaryAustin1** [reply]
You should contact support.  Not sure if you are saying you expect it to charge the extra $10 or you don't think you are at 500.  I believe you have to do something (I don't have a Pro plan) to add the extra "capacity" and it is not automatic.

My understanding is a connection is a websocket, and supabase-js will share that websocket for multiple subscriptions... if it matters.

**Reply by WasinWatt** [reply]
I've sent a support request yesterday but no response so far. I meant I expect it to charge the extra $10 because there is a lot of concurrent users in the past few days.

**Reply by ioucyf** [reply]
From [supabase.com](https://supabase.com/pricing) FAQ:

> "Can I cap my usage so my bill doesn't run over?
Yes. Spend caps are on by default on the pro tier. You can turn spend caps off for usage beyond the tier limits to pay as you grow."

I guess the cap setting (being on by default) is what's preventing you from going above your plan's limit.

**Reply by WasinWatt** [reply]
Yes, I did turn the spend cap off a few days ago but we still got `too_many_connections` anyway.

**Comment by peirix** [comment]
Did you figure this out? I'm getting the same, and in my testing it has never surpassed 150 connections.

**Comment by mark-at-pieces** [comment]
Hey all, any resolutions from this?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/realtime/rate-limits
There are also several different kind of rate errors.  It mentions checking the logs for more info, but I personally have not done that so not sure what info it provides.

**Reply by mark-at-pieces** [reply]
Update: reached out to support yesterday and got and instant response as well as a complete solution and a quota increase all in under 24 hours. Hats off to the Support + RealTime Team!

**Comment by chasers** [comment, marked as answer]
Hey all ... Chase from the Realtime team here. 

All these rate limits are adjustable and we are making them higher by default now that we've been running the new Realtime infra in prod for a while. We will still have rate limiting in place though to protect the cluster.

If you are hitting these we just need to change a config on our end to up it for you. Please contact support and we'll get it handled!!  And if you have an idea of the growth you're forecasting for the next ~12 months we will set you at that with lots of headroom.

**Comment by Xchanger25** [comment]
Built a small thing for this if it's useful.. watches four counters and emails
once when one spikes: https://github.com/Xchanger25/watchbill
